Docker.md

```diff
@@ -0,0 +1,171 @@
+
+Once all tools are installed, you will have two new executables that you can run in your terminal:
+1. `docker` - This is the application used for building and running containers
+2. `docker-compose` - [Docker Compose](https://docs.docker.com/compose/) is a tool for defining how to run any number of containers
+
+You can check that both of these applications are installed by running the following commands in your terminal:
+
+```
+$ which docker
+/usr/local/bin/docker
+
+$ which docker-compose
+/usr/local/bin/docker-compose
+```
+
+This uses the UNIX [which](https://www.tutorialspoint.com/unix_commands/which.htm) command to print the absolute path to the executable in question. As long as a path is printed for both `docker` and `docker-compose` then you are good to go - the paths do not need to be the same as those shown above.
+
+### Building
+
+You can build the Docker image for the `tbb` application by running `make image` in the root of this repository. This will take a minute or two to build and the output in your terminal will look something like this:
+
+```
+$ make image
+docker build -t tbb:latest .
+[+] Building 36.0s (9/9) FINISHED
+ => [internal] load build definition from Dockerfile                               0.0s
+ => => transferring dockerfile: 137B                                               0.0s
+ => [internal] load .dockerignore                                                  0.0s
+ => => transferring context: 2B                                                    0.0s
+ => [internal] load metadata for docker.io/library/golang:1.16                     0.4s
+ => [internal] load build context                                                  0.2s
+ => => transferring context: 6.78MB                                                0.2s
+ => [1/4] FROM docker.io/library/golang:1.16@sha256:8f29258b4b992b383d03290acde77  0.0s
+ => CACHED [2/4] WORKDIR /build                                                    0.0s
+ => [3/4] ADD ./ /build                                                            0.1s
+ => [4/4] RUN make install                                                        33.0s
+ => exporting to image                                                             2.3s
+ => => exporting layers                                                            2.3s
+ => => writing image sha256:b088ed645ed132544fe8257284a5ed03961da0801016a5544a5a6  0.0s
+ => => naming to docker.io/library/tbb:latest                                      0.0s
+
+Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
+```
+
+You can verify that the image was built successfully by running the command `docker images` in your terminal:
+
+```
+$ docker images
+REPOSITORY               TAG       IMAGE ID       CREATED          SIZE
+tbb                      latest    b088ed645ed1   15 seconds ago   1.12GB
+```
```

Comment:
I am seriously astonished by the level of effort and attention to detail you put into this PR.